localstack/dns/server.py

```diff
@@ -779,29 +779,22 @@ def get_available_dns_server():
 
 
 # ###### LEGACY METHODS ######
-def add_resolv_entry():
-    from localstack.services.edge import ensure_can_use_sudo
-
-    resolv_conf = "/etc/resolv.conf"
-    if os.path.exists(resolv_conf):
-        content = load_file(resolv_conf)
-
-        comment = "# The following line is required by LocalStack"
-        line = "nameserver 127.0.0.1"
-        if line not in content:
-            sudo_cmd = "" if is_root() else "sudo"
-            ensure_can_use_sudo()
-
-            for new_line in ("", line, comment):
-                # Surprisingly hard to find a reliable cross-platform shell
-                # solution to prepend text to a file, hence we use python here.
-                run(
-                    (
-                        '%s python -c "import sys; f=open(sys.argv[1]).read(); '
-                        "open(sys.argv[1], 'w').write('%s\\n' + f)\" %s"
-                    )
-                    % (sudo_cmd, new_line, resolv_conf)
-                )
+def add_resolv_entry(file_path: Path | str = Path("/etc/resolv.conf")):
+    # never overwrite the host configuration without the user's permission
+    if not in_docker():
+        LOG.warning("Incorrectly attempted to alter host networking config")
+        return
+
+    LOG.debug("Overwriting container DNS server to point to localhost")
+    content = textwrap.dedent(
+        """
+    # The following line is required by LocalStack
+    nameserver 127.0.0.1
+    """
+    )
+    file_path = Path(file_path)
+    with file_path.open("w") as outfile:
+        print(content, file=outfile)
```

Comment:
What if we are not running as root in the container? Probably sudo would not work then as well, but we should consider it, at least to handle the error correctly.

Reply:
Sure, that makes sense, thanks. I'll add an exception handler to at least catch any error that occurs.